health.go

```diff
@@ -0,0 +1,56 @@
+package grpc
+
+import proto "github.com/golang/protobuf/proto"
+import (
+	"time"
+
+	"golang.org/x/net/context"
+)
+
+type HealthCheckRequest struct {
```

Comment:
This all looks like generated protobuf code. Why isn't this in a .pb.go file?

If it's to avoid a circular dependency, the only thing here should be the service desc. But I don't understand why so much of this is exported either. This all needs more commentary.